### PR TITLE
[SYCL] Implement latest version of sycl_ext_oneapi_free_function_queries

### PR DESCRIPTION
**Pull request by aelovikov-intel**
- Implement latest revision of the proposed extension
- Deprecate `sycl::ext::oneapi::experimental::` interfaces
- Remove old deprecated interfaces from `sycl::` namespace

### CODE REVIEW

File: sycl/test/basic_tests/free_function_queries/free_function_queries_interface.cpp [hunk 1]
```diff
@@ -12,66 +12,36 @@
...
-  test<sycl::group>(this_group_caller<1>{});
-  test<sycl::group>(this_group_caller<2>{});
-  test<sycl::group>(this_group_caller<3>{});
+  test<sycl::group>(get_work_group_caller<1>{});
+  test<sycl::group>(get_work_group_caller<2>{});
+  test<sycl::group>(get_work_group_caller<3>{});
 
   static_assert(
-      std::is_same<decltype(sycl::ext::oneapi::experimental::this_sub_group()),
+      std::is_same<decltype(sycl::ext::oneapi::this_work_item::get_sub_group()),
```
Comment (bader) [inline sycl/test/basic_tests/free_function_queries/free_function_queries_interface.cpp:44]: is_save -> is_same_v?

**Review by Alcpz — APPROVED** [review]

File: sycl/include/sycl/ext/oneapi/experimental/root_group.hpp [hunk 2]
```diff
@@ -91,7 +92,8 @@ template <int Dimensions> sycl::sub_group get_child_group(group<Dimensions> g) {
 
 namespace this_kernel {
 template <int Dimensions> root_group<Dimensions> get_root_group() {
-  return this_nd_item<Dimensions>().ext_oneapi_get_root_group();
+  return sycl::ext::oneapi::this_work_item::this_nd_item<Dimensions>()
+      .ext_oneapi_get_root_group();
 }
 } // namespace this_kernel
```
Comment (Pennycook) [inline sycl/include/sycl/ext/oneapi/experimental/root_group.hpp:98]: Just a note: We should update the `root_group` extension, too.  This function should be defined as `this_work_item::get_root_group()` to match the others.

File: sycl/include/sycl/ext/oneapi/free_function_queries.hpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+#include <sycl/sub_group.hpp>
+
+// For deprecated queries:
+#include <sycl/id.hpp>
+#include <sycl/item.hpp>
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi::this_work_item {
+template <int Dimensions> nd_item<Dimensions> this_nd_item() {
```
[the remark this answers is not in the document]
Comment (aelovikov-intel) [reply sycl/include/sycl/ext/oneapi/free_function_queries.hpp]: Whoops... Pushing in the fix.

**Review by gmlueck — APPROVED** [review]

**Review by v-klochkov — APPROVED** [review]

File: sycl/test-e2e/Basic/free_function_queries/free_function_queries.cpp [hunk 4]
```diff
@@ -147,11 +147,12 @@ int main() {
...
+              sycl::ext::oneapi::this_work_item::get_nd_item<1>();
           results_acc[0] = that_nd_item == nd_i;
           auto nd_item_group = that_nd_item.get_group();
           results_acc[1] =
-              nd_item_group == sycl::ext::oneapi::experimental::this_group<1>();
+              nd_item_group ==
+              sycl::ext::oneapi::this_work_item::get_work_group<1>();
           auto nd_item_id = that_nd_item.get_global_id();
           results_acc[2] =
               nd_item_id == sycl::ext::oneapi::experimental::this_id<1>();
```
Comment (KseniyaTikhomirova) [inline sycl/test-e2e/Basic/free_function_queries/free_function_queries.cpp:158]: should it be updated too?

**Review by KseniyaTikhomirova — APPROVED** [review]
LGTM

File: sycl/test-e2e/Basic/free_function_queries/free_function_queries.cpp (hunk 4 above)
Comment (aelovikov-intel) [reply sycl/test-e2e/Basic/free_function_queries/free_function_queries.cpp:158]: Yes, thanks!

Comment (KseniyaTikhomirova) [reply sycl/test-e2e/Basic/free_function_queries/free_function_queries.cpp:158]: should it be removed then?

Comment (aelovikov-intel) [reply sycl/test-e2e/Basic/free_function_queries/free_function_queries.cpp:158]: The extension specification marks them as deprecated without providing an alternative. That is part of the change in this PR. We will only be able to remove 2025.1 or later.

**Pull request merged**